Agent version (#6129)
DataDog/documentation: content/en/agent/guide/agent-v6-python-3.md

---
title: Use Python 3 with Datadog Agent v6
kind: guide
disable_toc: true
further_reading:
- link: "agent/versions/upgrade_to_agent_v7"
  tag: "Documentation"
  text: "Upgrade to Agent v7"
---

Starting with v6.14.0, the Agent v6 integrates both Python 2 and Python 3 runtimes. This means that Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.

By default, the Agent v6 uses the Python 2 runtime. To switch to the Python 3 runtime:

{{< tabs >}}
{{% tab "Host Agent" %}}

1. Set the `python_version` configuration option [in your `datadog.yaml` configuration file][1]:

    ```yaml
    python_version: 3
    ```

2. [Restart the Agent][2].

Alternatively, the `DD_PYTHON_VERSION` environment variable can be set to `2` or `3` to choose which Python runtime is used. If it is set, the `python_version` option in `datadog.yaml` is ignored.

This is an Agent-wide configuration option: **all Python checks launched by an Agent use the same Python runtime**.

[1]: /agent/guide/agent-configuration-files/#agent-main-configuration-file
[2]: /agent/guide/agent-commands/#restart-the-agent
{{% /tab %}}
{{% tab "Containerized Agent" %}}

> As the only change between Agent v6.x and Agent v7.x is that the Agent v7.x includes the Python 3 runtime, contrary to Agent v6.x which include only the Python 2 runtime, to switch from one python runtime to another, choose the appropriate Agent image:

Reviewer: ```suggestion
Since the only change between Agent v6.x and Agent v7.x is that the Agent v7.x includes the Python 3 runtime, contrary to Agent v6.x which includes only the Python 2 runtime, you can switch Python runtimes by switching Agent versions. To switch from one Python runtime to another, choose the appropriate Agent image:
```

---
title: Use Python 3 with Datadog Agent v6
kind: guide
disable_toc: true
further_reading:
- link: "agent/versions/upgrade_to_agent_v7"
  tag: "Documentation"
  text: "Upgrade to Agent v7"
---

> Starting with v6.14.0, the Agent integrates both Python 2 and Python 3 runtimes. This means that Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.

Reviewer 2: ```suggestion
Starting with v6.14.0, the Agent v6 integrates both Python 2 and Python 3 runtimes. This means that Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.
```

By default, the Agent v6 uses the Python 2 runtime. To switch to the Python 3 runtime:

{{< tabs >}}
{{% tab "Host Agent" %}}

1. Set the `python_version` configuration option [in your `datadog.yaml` configuration file][1]:

    ```yaml
    python_version: 3
    ```

2. [Restart the Agent][2].

Alternatively, the `DD_PYTHON_VERSION` environment variable can be set to `2` or `3` to choose which Python runtime is used. If it is set, the `python_version` option in `datadog.yaml` is ignored.

This is an Agent-wide configuration option: **all Python checks launched by an Agent use the same Python runtime**.

[1]: /agent/guide/agent-configuration-files/#agent-main-configuration-file
[2]: /agent/guide/agent-commands/#restart-the-agent
{{% /tab %}}
{{% tab "Containerized Agent" %}}

> The official Containerized Agent images only include one of the two Python runtimes. To switch to one of the other runtime, choose the appropriate Agent image:

Reviewer: ```suggestion
The official Containerized Agent images only include one of the two Python runtimes. To switch from one to another runtime, choose the appropriate Agent image:
```

* **Python 2** runtime: Agent v6 images have the following format: `datadog/agent:<AGENT_VERSION>`, or `datadog/agent:<AGENT_VERSION>-jmx` for images supporting JMX checks.

> * **Python 3** runtime: Agent v6 images have the following format: `datadog/agent:<AGENT_VERSION>-py3`, or `datadog/agent:<AGENT_VERSION>-py3-jmx` for images supporting JMX checks.

Reviewer 2: there will be no more python 3 runtime image v6 Agent in the future, it's going to be replaced with the v7 Agent image as soon as v7 is GA (since it's functionally the same image)

Author: I updated the text but kept a clear separation to remind readers that agent v7 is for python 3 runtime and agent v6 is for Python 2 runtime
